docs/fernet.rst

```diff
@@ -109,6 +110,47 @@ has support for implementing key rotation via :class:`MultiFernet`.
     the front of the list to start encrypting new messages, and remove old keys
     as they are no longer needed.
 
+    .. method:: rotate(msg)
+
+        .. versionadded:: 2.2
+
+        Rotating tokens is necessary when any of the fernet keys of a
```

Comment:
I don't like this first sentence. Token rotation isn't just to limit the compromise window -- It's also for policy (rotation post-employee departure). Rotation in the event of actual compromise is a much more involved affair. Rotation as a best practice/cryptographic hygiene policy is designed to limit the damage in the event of an undetected event/increase the difficulty of attacks.

Reply:
@reaperhulk does this work a bit better?

```
        Rotates a token by re-encrypting it under the :class:`MultiFernet`
        instance's primary key. This preserves the timestamp that was originally
        saved with the token. If a token has successfully been rotated then the
        rotated token will be returned. If rotation fails this will raise an
        exception.

        Token rotation is a best practice and manner of cryptographic hygiene
        designed to limit damage in the event of an undetected event and to
        increase the difficulty of attacks. For example, it would be necessary
        to rotate all fernet tokens in the event of an employee leaving who
        previously had access to the fernet key used to encrypt/decrypt fernet
        tokens.
```

I initially added the sentence "rotating tokens is necessary..." to try provide a reason for someone to want to use `rotate`. I'm wondering if it might make more sense to pull the second part of the docstring I've proposed out of `rotate` and into the docs for `MultiFernet` itself.

Reply:
Yeah we have a section about key rotation in the multifernet class introduction so it probably does make sense to expand on it up there and link to the `rotate` method.